### PR TITLE
feat(encoding): impl `EncodeLabelValue` for `bool`

### PR DESCRIPTION
**Pull request by koushiro**
- impl `EncodeLabelValue` for `bool`
- adjust order of some impls (just for better consistency and readability)

### CODE REVIEW

File: src/encoding.rs [hunk 1]
```diff
@@ -536,6 +537,12 @@ where
     }
 }
 
+impl EncodeLabelValue for bool {
+    fn encode(&self, encoder: &mut LabelValueEncoder) -> Result<(), std::fmt::Error> {
+        encoder.write_str(if *self { "true" } else { "false" })
+    }
+}
```
Comment (cratelyn) [inline src/encoding.rs:544]: for the record, i would support this addition, i think it's a pleasant and natural addition to the library.

that said, the reordering of `impl` blocks here is probably best left for a distinct pull request, as it introduces a lot of undue noise for reviewers to inspect.

**Review by mxinden — COMMENTED** [review]
Thank you. 

Can you please add a changelog entry to the commit adding the impl for `bool`.

Please keep the split into two commits.

**Comment by koushiro** [comment]
There are some clippy warnings introduced by rust 1.83 during CI check.
Fixed by #250

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,8 +18,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Implement `Atomic<u64>` for `AtomicU64` for gauges.
   See [PR 226].
 
-[PR 226]: https://github.com/prometheus/client_rust/pull/198
+- Implement `EnableLabelValue` for bool.
```
Comment (mxinden) [inline CHANGELOG.md:21]:
```suggestion
- Implement `EnableLabelValue` for `bool`.
```

**Review by mxinden — APPROVED** [review]
Thanks!

One nit pick. Otherwise ready to merge.

Mind updating to latest `master`, or giving me access to merge `master` into this pull request?

**Pull request merged**